src/modules/dashboard/components/Dialogs/ControlSafeDialog/TransactionTypesSection/ContractInteractionSection.tsx

```diff
@@ -161,32 +162,58 @@ const ContractInteractionSection = ({
   );
 
   const onContractChange = useCallback(
-    (contract: AnyUser) => {
+    async (contract: AnyUser) => {
       setIsLoadingABI(true);
       setFetchABIError('');
+
       if (selectedSafe && isAddress(contract.profile.walletAddress)) {
-        const contractPromise = fetchContractABI(
+        const contractABIData = await fetchContractABI(
           contract.profile.walletAddress,
           Number(selectedSafe.chainId),
         );
-        contractPromise.then((data) => {
-          setIsLoadingABI(false);
-          if (data) {
-            onContractABIChange(data);
-          } else {
-            setFetchABIError(formatMessage(MSG.fetchFailedError));
-          }
+
+        if (contractABIData) {
+          onContractABIChange(contractABIData);
+        } else {
+          setFetchABIError(formatMessage(MSG.fetchFailedError));
+        }
+
+        const contractName = await fetchContractName(
+          contract.profile.walletAddress,
+          Number(selectedSafe?.chainId),
+        );
+
+        setFieldValue(`transactions.${transactionFormIndex}.contract`, {
```

Comment:
I believe this can be simplified to:
``` js
setFieldValue(
   `transactions.${transactionFormIndex}.contract.profile.displayName`,
   contractName || 'Unknown contract',
);
```